### PR TITLE
Fix and refactor few task in libvirt_up.yml

### PR DESCRIPTION
**Pull request by reihl**
TestBos2: virt

Some of them was bug some of them change to be more ansible format and some of them changed to modules

### CODE REVIEW

File: roles/ocp_on_libvirt/tasks/setup.yml [hunk 1]
```diff
@@ -47,14 +49,21 @@
       options kvm-{{ proc_family }} ept=1
     dest: /etc/modprobe.d/kvm_nested.conf
     mode: "0644"
-  when: nested_kvm.stdout != 'Y'
+  when: _ool_nested_kvm != 'Y'
```
Comment (ramperher) [inline roles/ocp_on_libvirt/tasks/setup.yml:52]: Here you use `_ool_nested_kvm` but you defined it before as `ool_nested_kvm`, `_` is missed at the beginning

File: roles/ocp_on_libvirt/tasks/setup.yml [hunk 2]
```diff
@@ -47,14 +49,21 @@
...
+    - _ool_result.rc is defined
+    - _ool_result.stderr is not ansible.builtin.regex('^.*Module kvm_(amd|intel) is in use.*$')
+  register: _ool_result
+  loop:
+    - absent
+    - present
+  community.general.modprobe:
+    name: "kvm_{{ proc_family }}"
+    state: "{{ item }}"
+  when: _ool_nested_kvm != 'Y'
```
Comment (ramperher) [inline roles/ocp_on_libvirt/tasks/setup.yml:66]: Same

File: roles/ocp_on_libvirt/tasks/setup.yml [hunk 3]
```diff
@@ -20,22 +24,20 @@
       - virt-install
       - langpacks-en
       - glibc-all-langpacks
-      - "libvirt-python{% if ansible_python.version.major == 3 %}3{% endif %}"
-      - "python{% if ansible_python.version.major == 3 %}3{% endif %}-netaddr"
-      - "python{% if ansible_python.version.major == 3 %}3{% endif %}-lxml"
-      - "python{{ ansible_python.version.major }}-pip"
```
[the remark this answers is not in the document]
Comment (reihl) [reply roles/ocp_on_libvirt/tasks/setup.yml:26]: No, it'll not, I'm assuming that in this case there is python3-pip and python2-pip and all packages above is python3 and python

File: roles/ocp_on_libvirt/tasks/setup.yml [hunk 4]
```diff
@@ -20,22 +24,20 @@
...
+      - "libvirt-{{ _ool_python_packages_prefix }}"
+      - "{{ _ool_python_packages_prefix }}-netaddr"
+      - "{{ _ool_python_packages_prefix }}-lxml"
+      - "{{ _ool_python_packages_prefix }}-pip"
+      - "{{ _ool_python_packages_prefix }}-virtualenv"
 
 - name: Get processor family
   ansible.builtin.set_fact:
-    proc_family: >
+    ool_proc_family: >-
```
[the remark this answers is not in the document]
Comment (reihl) [reply roles/ocp_on_libvirt/tasks/setup.yml:35]: used in multiple locations (modprobe and copy content to files)

File: roles/ocp_on_libvirt/tasks/setup.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (reihl) [reply roles/ocp_on_libvirt/tasks/setup.yml:26]: Fixed, thanks!

File: roles/ocp_on_libvirt/tasks/setup.yml [hunk 5]
```diff
@@ -8,6 +8,10 @@
     name: https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
   when: ansible_distribution_major_version == '7'
 
+- name: Set python prefix for packages
+  ansible.builtin.set_fact:
+    ool_python_packages_prefix: "{{ (ansible_python.version.major == 3) | ternary('python3', 'python') }}"
```
[the remark this answers is not in the document]
Comment (reihl) [reply roles/ocp_on_libvirt/tasks/setup.yml]: Changed. Thanks!

File: roles/ocp_on_libvirt/tasks/setup.yml (hunk 1 above)
Comment (reihl) [reply roles/ocp_on_libvirt/tasks/setup.yml:53]: Fixed

**Comment by reihl** [comment]
retest

**Comment by reihl** [comment]
retest

File: roles/ocp_on_libvirt/tasks/setup.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (ramperher) [reply roles/ocp_on_libvirt/tasks/setup.yml:26]: thanks! closing this

File: roles/ocp_on_libvirt/tasks/setup.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (ramperher) [reply roles/ocp_on_libvirt/tasks/setup.yml:33]: ack, forgot that, closing this

**Review by ramperher — APPROVED** [review]
recheck

**Comment by ramperher** [comment]
recheck

**Comment by ramperher** [comment]
I've seen an issue with the merge-queue job, I'll try to fix it separately, now that I've identified that.

**Comment by ramperher** [comment]
> I've seen an issue with the merge-queue job (not related to this change, but related to the code that the merge queue action has), I'll try to fix it separately, now that I've identified that.

Created this change to fix the issue with dci-merge: https://github.com/redhatci/ansible-collection-redhatci-ocp/pull/334, please hold on this change while that PR is reviewed, I'd prefer to fix that before trying to re-merge this to confirm the issue is fixed.

**Comment by ramperher** [comment]
Found another issue with dci-merge, fixing it here: https://github.com/redhatci/ansible-collection-redhatci-ocp/pull/335

**Comment by ramperher** [comment]
Change was merged, trying to merge this again.

**Pull request merged**